aws/resource_aws_s3_bucket_replication_configuration.go

```diff
@@ -0,0 +1,633 @@
+
+		if ssc, ok := rr["source_selection_criteria"].([]interface{}); ok && len(ssc) > 0 {
+			if ssc[0] != nil {
+				sscValues := ssc[0].(map[string]interface{})
+				ruleSsc := &s3.SourceSelectionCriteria{}
+				if sseKms, ok := sscValues["sse_kms_encrypted_objects"].([]interface{}); ok && len(sseKms) > 0 {
+					if sseKms[0] != nil {
+						sseKmsValues := sseKms[0].(map[string]interface{})
+						sseKmsEncryptedObjects := &s3.SseKmsEncryptedObjects{}
+						sseKmsEncryptedObjects.Status = aws.String(sseKmsValues["status"].(string))
+						ruleSsc.SseKmsEncryptedObjects = sseKmsEncryptedObjects
+					}
+				}
+				if sscRm, ok := sscValues["replica_modifications"].([]interface{}); ok && len(sscRm) > 0 {
+					if sscRm[0] != nil {
+						replicaModValues := sscRm[0].(map[string]interface{})
+						replicaModifications := &s3.ReplicaModifications{}
+						replicaModifications.Status = aws.String(replicaModValues["status"].(string))
+						ruleSsc.ReplicaModifications = replicaModifications
+					}
+				}
+				rcRule.SourceSelectionCriteria = ruleSsc
+			}
+		}
+
+		if f, ok := rr["filter"].([]interface{}); ok && len(f) > 0 && f[0] != nil {
+			// XML schema V2.
+			rcRule.Priority = aws.Int64(int64(rr["priority"].(int)))
+			rcRule.Filter = &s3.ReplicationRuleFilter{}
+			filter := f[0].(map[string]interface{})
+			tags := keyvaluetags.New(filter["tags"]).IgnoreAws().S3Tags()
+			if len(tags) > 0 {
+				rcRule.Filter.And = &s3.ReplicationRuleAndOperator{
+					Prefix: aws.String(filter["prefix"].(string)),
+					Tags:   tags,
+				}
+			} else {
+				rcRule.Filter.Prefix = aws.String(filter["prefix"].(string))
+			}
+
+			dmr, ok := rr["delete_marker_replication"].([]interface{})
+			if ok && len(dmr) > 0 {
+				s := dmr[0].(map[string]interface{})
+				rcRule.DeleteMarkerReplication = &s3.DeleteMarkerReplication{
+					Status: aws.String(s["status"].(string)),
+				}
+			}
+		} else {
+			// XML schema V1.
+			rcRule.Prefix = aws.String(rr["prefix"].(string))
+		}
+
+		rules = append(rules, rcRule)
+	}
+
+	rc.Rules = rules
+	i := &s3.PutBucketReplicationInput{
+		Bucket:                   aws.String(bucket),
+		ReplicationConfiguration: rc,
+	}
+	log.Printf("[DEBUG] S3 put bucket replication configuration: %#v", i)
+
+	err := resource.Retry(1*time.Minute, func() *resource.RetryError {
+		_, err := s3conn.PutBucketReplication(i)
+		if isAWSErr(err, s3.ErrCodeNoSuchBucket, "") || isAWSErr(err, "InvalidRequest", "Versioning must be 'Enabled' on the bucket") {
+			return resource.RetryableError(err)
+		}
+		if err != nil {
+			return resource.NonRetryableError(err)
+		}
+		return nil
+	})
+	if isResourceTimeoutError(err) {
+		_, err = s3conn.PutBucketReplication(i)
+	}
+	if err != nil {
+		return fmt.Errorf("Error putting S3 replication configuration: %s", err)
+	}
+
+	return resourceAwsS3BucketReplicationConfigurationRead(d, meta)
+}
+
+func resourceAwsS3BucketReplicationConfigurationDelete(d *schema.ResourceData, meta interface{}) error {
+
```

Comment:
Hi @dkujawski 👋 I noticed this part hasn't been implemented. do want to add delete capability with https://docs.aws.amazon.com/AmazonS3/latest/API/API_DeleteBucketReplication.html ?

Reply:
Good catch! Yes, I do believe we want that to be included.  I can look at adding it, unless you already have a momentum in that area.  Let me know.

Reply:
Awesome, thanks! No work has been started on my end 😄 Let us know when its's ready for another look